src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,184 @@
+
+/** 
+ * An example subsystem.  You can replace me with your own Subsystem. 
+ */
+public class Lift extends Subsystem {
+    // Defines Limit Switches (Digital imputs)
+    DigitalInput TopLimitSwitchLift = new DigitalInput(RobotMap.DigitalImputs.LIMIT_SWITCH_LIFT_BOTTOM);
+    DigitalInput BottomLimitSwitchLift = new DigitalInput(RobotMap.DigitalImputs.LIMIT_SWITCH_LIFT_TOP);
+    // Defines Encoders 
+    Encoder liftEncoder = new Encoder(RobotMap.Encoders.LIFT_A, RobotMap.Encoders.LIFT_B);
+    // Defines Motors 
+    WPI_TalonSRX m_liftmotor = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_A);
+    WPI_TalonSRX m_liftmotor2 = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_B);
+    // Defines the previus motors as one motor 
+    SpeedControllerGroup liftdrive = new SpeedControllerGroup(m_liftmotor, m_liftmotor2);
+    DoubleSolenoid LiftTransmission = new DoubleSolenoid(RobotMap.Solenoids.LiftTransmission,
+            RobotMap.Solenoids.LiftTransmission2);
+
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        addChild(liftEncoder);
+        //will be changed
+        liftEncoder.setDistancePerPulse(.1);
+    }
+
+    private Boolean TopLimitSwitchOn() {
+        if (TopLimitSwitchLift.get())
+            return true;
+        else
+            return false;
+    }
+
+    private Boolean BottomLimitSwitchOn() {
+        if (BottomLimitSwitchLift.get())
+            return true;
+        else
+            return false;
+    }
+
+    //gets the Lift Encoder's distance
+    public double EncoderDistance = liftEncoder.getDistance();
+    //gets the Lift Encoder's speed
+    public double EncoderSpeed = liftEncoder.getRate();
+
+    //sets the motor's speed, can be called later
+    public void goUp() {
+        liftdrive.set(0.25);
+    }
+
+    public void goDown() {
+        liftdrive.set(-0.25);
+    }
+
+    //gear changes
+    public void shiftToHighGear() {
+        if (LiftTransmission.get() != Value.kForward) {
+            LiftTransmission.set(Value.kForward);
+        }
+    }
+
+    public void shiftToLowGear() {
+        if (LiftTransmission.get() != Value.kReverse) {
+            LiftTransmission.set(Value.kReverse);
+        }
+    }
+
+    public void initDefaultCommand() {
+        setDefaultCommand(new SubsystemCommand(this) {
+            @Override
+            protected boolean isFinished() {
+                return false;
+            }
+        });
+    }
+
+    public Command GoToHeight(LiftHeight height) {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                shiftToHighGear();
```

Comment:
Will this work when the climb command runs?

Reply:
Agreed, this should either be changed to work specifically for high gear (rename this command) or take a parameter that let's you call out which gear you want.